common/src/main/kotlin/streams/service/errors/KafkaErrorService.kt

```diff
@@ -1,28 +1,35 @@
 package streams.service.errors
 
 import org.apache.commons.lang3.exception.ExceptionUtils
+import org.apache.kafka.clients.CommonClientConfigs
 import org.apache.kafka.clients.producer.KafkaProducer
 import org.apache.kafka.clients.producer.Producer
 import org.apache.kafka.clients.producer.ProducerConfig
 import org.apache.kafka.clients.producer.ProducerRecord
 import org.apache.kafka.common.record.RecordBatch
 import org.apache.kafka.common.serialization.ByteArraySerializer
 import org.neo4j.util.VisibleForTesting
+import streams.utils.ValidationUtils.validateConnection
 import java.util.*
 
 class KafkaErrorService(private val producer: Producer<ByteArray, ByteArray>?, private val errorConfig: ErrorConfig, private val log: (String, Exception?)->Unit): ErrorService() {
 
     constructor(config: Properties, errorConfig: ErrorConfig,
-                log: (String, Exception?) -> Unit) : this(producer(errorConfig, config), errorConfig, log)
+                log: (String, Exception?) -> Unit) : this(producer(errorConfig, config, log), errorConfig, log)
 
     companion object {
-        private fun producer(errorConfig: ErrorConfig, config: Properties) =
+        private fun producer(errorConfig: ErrorConfig, config: Properties, log: (String, Exception?) -> Unit) =
                 errorConfig.dlqTopic?.let {
-                    val props = Properties()
-                    props.putAll(config)
-                    props[ProducerConfig.KEY_SERIALIZER_CLASS_CONFIG] = ByteArraySerializer::class.java.name
-                    props[ProducerConfig.VALUE_SERIALIZER_CLASS_CONFIG] = ByteArraySerializer::class.java.name
-                    KafkaProducer<ByteArray,ByteArray>(props)
+                    try {
+                        val bootstrapServers = config.getOrDefault(CommonClientConfigs.BOOTSTRAP_SERVERS_CONFIG, "").toString()
+                        validateConnection(bootstrapServers, CommonClientConfigs.BOOTSTRAP_SERVERS_CONFIG, false)
+                        config[ProducerConfig.KEY_SERIALIZER_CLASS_CONFIG] = ByteArraySerializer::class.java.name
```

Comment:
what happens if someone wants to use AVRO for their DLQ?
Are we overriding things then? No need to fix now but wee should probably be aware and document this?
Esp. if their DLQ for Kafka Connect itself was configured with AVRO and the same topic it might break stuff?

Reply:
at this moment we return in the dlq only the content of the AVRO event without the schema as ByteArray, we can improve this a future PR with the support for custom serializer also for the DLQ.